### PR TITLE
bump(github.com/opencontainers/runc/libcontainer)

### PR DESCRIPTION
**Pull request by jimmidyson**
Fixes issues with breaking changes to `GetPids` which is affecting
downstream consumers of cadvisor (e.g. Kubernetes).

Early PR to test functionality. Please don't merge until new libcontainer tag is
applied.

### CODE REVIEW

**Comment by jimmidyson** [comment]
@pwittrock @vishh Problems with Jenkins e2e:

```
F0112 12:48:17.258015   30970 runner.go:246] Error 0: failed to make remote testing directory: command "gcloud" ["compute" "ssh" "--zone" "us-central1-f" "e2e-cadvisor-rhel-7-docker19" "--" "mkdir" "-p" "/tmp/cadvisor-30970"] failed with error: exit status 1 and output: "ERROR: (gcloud.compute.ssh) Your current active account [211744435552-compute@developer.gserviceaccount.com] does not have any valid credentials\nPlease run:\n\n  $ gcloud auth login\n\nto obtain new credentials, or if you have already logged in with a\ndifferent account:\n\n  $ gcloud config set account ACCOUNT\n\nto select an already authenticated account to use.\n"
Error 1: failed to make remote testing directory: command "gcloud" ["compute" "ssh" "--zone" "us-central1-f" "e2e-cadvisor-container-vm-v20151215-docker18" "--" "mkdir" "-p" "/tmp/cadvisor-30970"] failed with error: exit status 1 and output: "ERROR: (gcloud.compute.ssh) Your current active account [211744435552-compute@developer.gserviceaccount.com] does not have any valid credentials\nPlease run:\n\n  $ gcloud auth login\n\nto obtain new credentials, or if you have already logged in with a\ndifferent account:\n\n  $ gcloud config set account ACCOUNT\n\nto select an already authenticated account to use.\n"
Error 2: failed to make remote testing directory: command "gcloud" ["compute" "ssh" "--zone" "us-central1-f" "e2e-cadvisor-coreos-beta-docker19" "--" "mkdir" "-p" "/tmp/cadvisor-30970"] failed with error: exit status 1 and output: "ERROR: (gcloud.compute.ssh) Your current active account [211744435552-compute@developer.gserviceaccount.com] does not have any valid credentials\nPlease run:\n\n  $ gcloud auth login\n\nto obtain new credentials, or if you have already logged in with a\ndifferent account:\n\n  $ gcloud config set account ACCOUNT\n\nto select an already authenticated account to use.\n"
```

Can someone fix that please as I don't want to merge this until I've got successful e2e tests at least.

**Comment by jimmidyson** [comment]
@k8s-bot retest this please

**Comment by jimmidyson** [comment]
@pwittrock @vishh Also how can I retrigger Jenkins GCE e2e tests?

**Comment by pwittrock** [comment]
'retest this please' should have worked.  We are just using the vanilla [jenkins pr builder plugin] (https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+request+builder+plugin)  Closing and reopening should also work if nothing else is

**Comment by pwittrock** [comment]
retest this please

**Comment by pwittrock** [comment]
fwiw: The gcloud auth issue is something I have been trying to track down for the past week+.  It seems to be flaky and unrelated to whether or not there are actual gcloud credentials set up.  I have a PR out to try and get more debug info in our k8s e2e test where we also see this happening.

**Comment by pwittrock** [comment]
retest this please

**Comment by pwittrock** [comment]
'retest this please' does not appear to be working for me either.  I'll have to dig into the Jenkins logs to see if I can debug what the underlying issue is.  As a work around pushing any code change should trigger the test, or closing/reopening the issue should also work.

**Comment by jimmidyson** [comment]
OK thanks. Didn't think closing/opening issue worked with ghprb? Thought build result was tied to commit sha?

**Comment by pwittrock** [comment]
Hm, I thought I remember triggering builds in the past by closing/reopening, but something else could have been going on that was triggering the build.

**Comment by pwittrock** [comment]
@k8s-bot retest this please

**Comment by pwittrock** [comment]
test this please

**Comment by pwittrock** [comment]
@k8s-bot retest this please

**Comment by pwittrock** [comment]
test this please

**Comment by pwittrock** [comment]
@k8s-bot test this

**Comment by pwittrock** [comment]
I manually added the trigger phrase "@k8s-bot test this" to the ghprb configuration, and that seems to be working now.  Not sure why the built in commands do not work.

**Comment by jimmidyson** [comment]
@pwittrock Thank you very much for looking at that.

& e2e is green!

**Comment by jimmidyson** [comment]
Still please don't merge until runc is tagged & we can reference a tagged version in godeps.

**Comment by jimmidyson** [comment]
@vishh @pwittrock This is now ready to merge so we can fix up some Kubernetes issues.

File: container/docker/handler.go [hunk 1]
```diff
@@ -206,20 +206,23 @@ func libcontainerConfigToContainerSpec(config *libcontainerconfigs.Config, mi *i
...
-	spec.HasCpu = true
-	spec.Cpu.Limit = 1024
-	if config.Cgroups.CpuShares != 0 {
-		spec.Cpu.Limit = uint64(config.Cgroups.CpuShares)
+	if config.Cgroups.Resources != nil {
+		if config.Cgroups.Resources.Memory > 0 {
+			spec.Memory.Limit = uint64(config.Cgroups.Resources.Memory)
+		}
+		if config.Cgroups.Resources.MemorySwap > 0 {
+			spec.Memory.SwapLimit = uint64(config.Cgroups.Resources.MemorySwap)
```
Comment (vishh) [inline container/docker/handler.go]: Why do we need an explicit cast here and above?

**Comment by vishh** [comment]
Just one comment. Otherwise LGTM.

File: container/docker/handler.go (hunk 1 above)
Comment (jimmidyson) [inline container/docker/handler.go]: `config.Cgroups.Resources.Memory` is int64, `spec.Memory.Limit` is uint64

Comment (vishh) [inline container/docker/handler.go]: Thats weird. The kernel expects it to be a `uint64`. So thats an issue against `runc` I guess. 
LGTM then.

Comment (jimmidyson) [inline container/docker/handler.go]: Very weird.

**Comment by jimmidyson** [comment]
And now I find out they're redoing the runc tag as they missed a version bump in code... Will update the dep when that's done in a new PR.

**Pull request merged**